src/internet_identity/src/stats/event_stats.rs

```diff
@@ -420,21 +424,18 @@ fn update_aggregation<M: Memory, F>(
     pruned_weight_by_key
         .into_iter()
         .for_each(|(key, pruned_weight)| {
-            let current_weight = db.get(&key).unwrap_or_else(|| {
-                panic!("aggregation key \"{:?}\" not found in DB when pruning", key)
-            });
-            assert!(
-                current_weight >= pruned_weight,
-                "pruned weight {} exceeds current weight {} for key \"{:?}\"",
-                pruned_weight,
-                current_weight,
-                key
-            );
-            let new_weight = current_weight - pruned_weight;
-            if new_weight == 0 {
-                db.remove(&key);
+            // Adjust aggregation weight by subtracting the pruned weight
+            if let Some(current_weight) = db.get(&key) {
+                let new_weight = current_weight.saturating_sub(pruned_weight);
+                if new_weight == 0 {
+                    db.remove(&key);
+                } else {
+                    db.insert(key, new_weight);
+                }
             } else {
-                db.insert(key, new_weight);
+                // This should not happen, but if it does, it's a bug.
+                // Print a message so that we can investigate once canister logs are available.
+                ic_cdk::println!("WARN: Aggregation key {:?} not found in db", key);
```

Comment:
Is this `key` a public key? @frederikrothenberger

Reply:
@cybrowl: No, this is a key into a map of event aggregations that we use to collect usage statistics of II. See [here](https://github.com/dfinity/internet-identity/blob/8d2edeb81d59e77ac85d4421a48c22c09d9db8c7/src/internet_identity/src/stats/event_stats/event_aggregations.rs#L256).